x-pack/plugins/task_manager/server/MONITORING.md

```diff
@@ -0,0 +1,242 @@
+  - The average _drift_ that tasks experience (mean and median, using the same configurable running average window as above). Drift tells us how long after a task's scheduled a task typically executes.
+  - The polling rate (the timestamp of the last time a polling cycle completed) and the result [`No tasks | Filled task pool | Unexpectedly ran out of workers`] frequency the past 50 polling cycles (using the same window size as the one used for running averages)
+  - The `Success | Retry | Failure ratio` by task type. This is different than the workload stats which tell you what's in the queue, but ca't keep track of retries and of non recurring tasks as they're wiped off the index when completed.
+
+These are "Hot" stats which are updated reactively as Tasks are executed and interacted with.
+
+### Example Stats
+
+For example, if you _curl_ the `/api/task_manager/_health` endpoint, you might get these stats:
+```
+{
+	 /* the time these stats were returned by the api */
+	"timestamp": "2020-10-05T18:26:11.346Z",
+	 /* the overall status of the system */
+	"status": "OK",
+	 /* last time any stat was updated in this output */
+	"lastUpdate": "2020-10-05T17:57:55.411Z",    
+	"stats": {
+		"configuration": {      /* current configuration of TM */
+			"timestamp": "2020-10-05T17:56:06.507Z",
+			"status": "OK",
+			"value": {
+				"max_workers": 10,
+				"poll_interval": 3000,
+				"request_capacity": 1000,
+				"max_poll_inactivity_cycles": 10,
+				"monitored_aggregated_stats_refresh_rate": 60000,
+				"monitored_stats_running_average_window": 50
+			}
+		},
+		"workload": {  /* The workload of this deployment */
+			"timestamp": "2020-10-05T17:57:06.534Z",
+			"status": "OK",
+			"value": {
+				"count": 6,        /* count of tasks in the system */
+				"taskTypes": {   /* what tasks are there and what status are they in */
+					"actions_telemetry": {
+						"count": 1,
+						"status": {
+							"idle": 1
+						}
+					},
+					"alerting_telemetry": {
+						"count": 1,
+						"status": {
+							"idle": 1
+						}
+					},
+					"apm-telemetry-task": {
+						"count": 1,
+						"status": {
+							"idle": 1
+						}
+					},
+					"endpoint:user-artifact-packager": {
+						"count": 1,
+						"status": {
+							"idle": 1
+						}
+					},
+					"lens_telemetry": {
+						"count": 1,
+						"status": {
+							"idle": 1
+						}
+					},
+					"session_cleanup": {
+						"count": 1,
+						"status": {
+							"idle": 1
+						}
+					}
+				},
+
+				/* Frequency of recurring tasks schedules */
+				"schedule": [  
+					["60s", 1],   /* 1 task, every 60s */
+					["3600s", 3],  /* 3 tasks every hour */
+					["720m", 1]
+				],
+				/* There are no overdue tasks in this system at the moment */
+				"overdue": 0, 
+				/* This is the schedule density, it shows a histogram of all the  polling intervals in the next minute (or, if 
```

Comment:
nit: these comments overflow GH's text box width, making it hard to see the everything at once.  8 char indenting is likely making it "worse", heh.  But keep that at _at least_ 4 so we can see some of the structure easier :-)